core/src/main/java/org/apache/james/core/MailAddress.java

```diff
@@ -129,6 +129,23 @@ public static  MailAddress getMailSender(String sender) {
         }
     }
 
+    public static MailAddress asMailAddress(String address) {
+        try {
+            return new MailAddress(address);
+        } catch (AddressException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static Optional<MailAddress> asMailAddressOptional(String address) {
+        try {
+            return Optional.of(new MailAddress(address));
+        } catch (AddressException e) {
+            LOGGER.debug("Can't parse address {}", address, e);
+            throw new RuntimeException(e);
```

Comment:
you mean return Optional.empty()? meaning you probably miss some tests?
What happens if address is null by the way?

Reply:
A stupid mistake on my side, that our test suite caught...

I gonna add the tests for these static methods